### PR TITLE
pagefind: 1.1.0 -> 1.1.1

### PR DESCRIPTION
**Pull request by pbsds**
This has got to be one of the more cursed vendorings I've done recently. [I exhausted every other option](https://github.com/pbsds/nixpkgs/commit/35e14c5100df).

Diff: https://github.com/CloudCannon/pagefind/compare/v1.1.0...v1.1.1
Release: https://github.com/CloudCannon/pagefind/releases/tag/v1.1.1
Changelog: https://github.com/CloudCannon/pagefind/blob/v1.1.1/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
you can test it with 
```
nix shell github:nixos/nixpkgs/refs/pull/344950/merge#pagefind
```

File: pkgs/applications/misc/pagefind/default.nix [hunk 1]
```diff
@@ -88,13 +106,20 @@ rustPlatform.buildRustPackage rec {
...
 
   # build wasm and js assets
   # based on "test-and-build" in https://github.com/CloudCannon/pagefind/blob/main/.github/workflows/release.yml
   preBuild = ''
     export HOME=$(mktemp -d)
 
+    # serve lindera-unidic on localhost
+    mkdir .lindera-http-plz
+    ln -s ${lindera-unidic-src} .lindera-http-plz/unidic-mecab-2.1.2.tar.gz
+    httplz --port 34567 -- .lindera-http-plz/ &
```
Comment (phanirithvij) [inline pkgs/applications/misc/pagefind/default.nix:120]: Does it make sense to hardcode 34567? can we use a port assigned by the os? i.e. 0
Then serve the file using httplz in postPatch before substituing the url?

sent a pr, https://github.com/pbsds/nixpkgs/pull/1

Comment (pbsds) [reply pkgs/applications/misc/pagefind/default.nix:120]: It is safe within the linux sandbox, but not so much on darwin. i'll look at the PR

**Comment by pbsds** [comment]
I rebased, reworded and adapted your commit.

**Review by phanirithvij — APPROVED** [review]
tested with nix-community/noogle locally, LGTM

**Pull request merged**